File: BACKEND/BE/Models/Tarea.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace BE.Models
{
    public class Tarea
    {
        public int ID { get; set; }

        public string Descripcion { get; set; } = null!;

        public DateTime FechaHora { get; set; }

        public string Ubicacion { get; set; } = null!;
        //public virtual ICollection<Voluntario> Voluntarios { get; set; }

  

        [ForeignKey("Evento")]
        public int EventoID { get; set; }

        public virtual Evento Evento { get; set; }
        public virtual ICollection<TareaVoluntario> TareaVoluntarios { get; set; }


    }
}

File: FRONTEND/FE-CRUDEventos/src/app/interfaces/tarea.ts
export interface Voluntario {
  id: number;
  email: string;
  nombre: string;
  apellido: string;
  telefono: string;
}

export interface TareaVoluntarios {
  tareaID: number;
  voluntarioID: string;
  estado: string;
  comentario: string;
}

export interface Tarea {
  id: number;
  descripcion: string;
  fechaHora: Date;
  ubicacion: string;
  tareaVoluntarios: TareaVoluntarios[];
}

export interface CreateTarea {
  // estado: string;
  descripcion: string;
  fechaHora: Date;
  ubicacion: string;
  eventoID: number;
  voluntarios: Voluntario[];
}
